pkg/domain/entities/types/containers.go

```diff
@@ -36,7 +37,42 @@ type ContainerStatsReport struct {
 }
 
 type ContainerUpdateOptions struct {
-	NameOrID                        string
+	NameOrID string
+	// Deprecated: Specgen should not be used to change the container configuration.
+	// Specgen is processed first, so values will be overwritten by values from other fields.
+	// To change configuration use other fields:
+	// Resources to change resource configuration, DevicesLimits to Limit device,
+	// RestartPolicy to change restart policy and RestartRetries to change restart retries.
 	Specgen                         *specgen.SpecGenerator
+	Resources                       *specs.LinuxResources
+	DevicesLimits                   *define.UpdateContainerDevicesLimits
 	ChangedHealthCheckConfiguration *define.UpdateHealthCheckConfig
+	RestartPolicy                   *string
+	RestartRetries                  *uint
```

Comment:
@mheon PTAL I never liked using specgen here as it is totally unclear what field are or are not supported so I am fine with this but I am not sure if have objects?

Reply:
We're not actually using the full thing, so passing a full Specgen feels deceptive. A lot easier to say what we actually support like this.

Reply:
Updated the doc string with the list of fields that are currently used from `SpecGenerator` struct.